src/uu/cksum/src/cksum.rs

```diff
@@ -134,8 +134,14 @@ where
         } else if filename.is_dir() {
             Box::new(BufReader::new(io::empty())) as Box<dyn Read>
         } else {
-            file_buf =
-                File::open(filename).map_err_context(|| filename.to_str().unwrap().to_string())?;
+            file_buf = match File::open(filename) {
+                Ok(file) => file,
+                Err(_) => {
+                    eprintln!("cksum: {}: No such file or directory", filename.display());
```

Comment:
I think you got the control flow right, but the error message would now be wrong if there is another reason that the file can't be opened, for example if the permissions are incorrect.

A more correct approach would look like this:
```rust
file_buf = match File::open(filename) {
    Ok(file) => file,
    Err(e) => {
        show!(e.map_err_context(|| filename.to_string_lossy().to_string());
        continue;
    }
};
```

This will automatically add "cksum" and print the error message corresponding to the io error that occurred.

See also the docs for
- [`map_err_context`](https://docs.rs/uucore/latest/uucore/error/trait.FromIo.html)
- [`show!`](https://docs.rs/uucore/latest/uucore/macro.show.html)

Reply:
Thank you for the kind gesture